PowerPointLabs/PowerPointLabs/SyncLab/View/SyncFormatDialog.xaml.cs

```diff
@@ -16,15 +16,26 @@ namespace PowerPointLabs.SyncLab.View
     public partial class SyncFormatDialog : Window
     {
 
+        string originalName;
         FormatTreeNode[] formats = null;
 
-        public SyncFormatDialog(Shape shape) : this(shape, SyncFormatConstants.FormatCategories)
+        public SyncFormatDialog(Shape shape) : this(shape, shape.Name, SyncFormatConstants.FormatCategories)
         {
         }
 
-        public SyncFormatDialog(Shape shape, FormatTreeNode[] formats)
+        public SyncFormatDialog(Shape shape, string formatName, FormatTreeNode[] formats)
         {
             InitializeComponent();
+            formatName = formatName.Trim();
+            if (SyncFormatUtil.IsValidFormatName(formatName))
```

Comment:
If the name is not a valid format, I will need to revert it to
1) The original text before it was changed
2) The original shape's text

This will differ based on where it is called from.